core/ui.py

```diff
@@ -307,22 +308,19 @@ def CurrentLocation(self):
   def PrefixPrint(self, msg, prefix, blame_loc):
     # type: (str, str, loc_t) -> None
     """Print a hard-coded message with a prefix, and quote code."""
-    span_id = location.GetSpanId(blame_loc)
-    _PrintWithSpanId(prefix, msg, span_id, self.arena, show_code=True)
-
-  def Print_(self, msg, span_id=runtime.NO_SPID):
-    # type: (str, int) -> None
-    """Print a hard-coded message, and quote code."""
-    if span_id == runtime.NO_SPID:
-      span_id = self.CurrentLocation()
-    _PrintWithSpanId('', msg, span_id, self.arena, show_code=True)
-
-  def PrintMessage(self, msg, span_id=runtime.NO_SPID):
-    # type: (str, int) -> None
+    _PrintWithSpanId(prefix, msg, blame_loc, self.arena, show_code=True)
+
+  def Print_(self, msg, blame_loc=None):
+    # type: (str, loc_t) -> None
+    if not blame_loc:
+        blame_loc = loc.Missing()
+    _PrintWithSpanId('', msg, blame_loc, self.arena, show_code=True)
+
+  def PrintMessage(self, msg):
```

Comment:
Hm interesting, looks like we do not have any PrintMessage() calls with explicit locations. 

But I think we should make this the same as `Print_` -- it will accept a  location, and fall back  on the current one if missing

Another step, doesn't need to be in this PR: I want to make blame_loc REQUIRED, so then we can grep for `loc.Missing()` to find the actual missing locations

We should probably go "wide" before we go "deep" into doing that